### PR TITLE
update tutorial on writing and running tests

### PR DESCRIPTION
**Pull request by jmikovic**
tutorial explains that tests shouldn't modify system,
how to prepare environment and how to run tests for actors and libs

### CODE REVIEW

File: docs/source/unit-testing.md [hunk 1]
```diff
@@ -86,19 +87,7 @@ still possible to use - see their [documentation](pydoc/leapp.html#module-leapp.
...
-as necessary and after executing the actor, revert back all the changes, so that
-other tests work with the same environment. This option may be quite challenging
-and in some cases not doable, for example when disabling SELinux which requires
-reboot.
-- replace the functions that read or modify the system with functions that do
-not alter the system and return what you specify in the test. This is called
-mocking. Currently it is not possible to mock any function while using the
-`current_actor_context.run()`. But, mocking is possible in an actor's library.
-For that, read further.
+Tests shouldn't modify the system in test.
```
Comment (bocekm) [inline docs/source/unit-testing.md]: I'd leave the sentence that was there in the original form or another, it doesn't matter:
```
- replace the functions that read or modify the system with functions that do
not alter the system and return what you specify in the test. This is called
mocking. Currently it is not possible to mock any function while using the
`current_actor_context.run()`. But, mocking is possible in an actor's library.
For that, read further.
```
The title of this section reads: "_Testing actors that modify the OS_". There are certainly actors that modify the system and this section should tell people how to test such actors.

File: docs/source/unit-testing.md [hunk 2]
```diff
@@ -155,8 +144,31 @@ shared libraries, models, etc.
...
+If your actor requires any package when executed as part of a workflow, it
+needs to be specified in a
+[leapp-repository specfile](https://github.com/oamg/leapp-repository/blob/master/packaging/leapp-repository.spec).
+
 ## Running the tests
 
+### Preparing the environment
+
+To execute unit tests of actors from all Leapp repositories in the
+`leapp-repository` GitHub repository, you need to install test dependencies for each
```
Comment (bocekm) [inline docs/source/unit-testing.md]:
```suggestion
`leapp-repository` GitHub repository, you need to install test dependencies for all
```

File: docs/source/unit-testing.md [hunk 3]
```diff
@@ -155,8 +144,31 @@ shared libraries, models, etc.
...
+actor by running  `make install-deps`.
```
Comment (bocekm) [inline docs/source/unit-testing.md]:
```suggestion
actors by running  `make install-deps`.
```

File: docs/source/unit-testing.md [hunk 4]
```diff
@@ -155,8 +144,31 @@ shared libraries, models, etc.
...
+
 ### Actor's tests
 
 Use _pytest_ to run the tests for your actor. For the test modules to be able to
```
Comment (bocekm) [inline docs/source/unit-testing.md]: I think you can remove the "manual" way using pytest and leave just there the `make test`.

**Comment by jmikovic** [comment]
@bocekm any reason it's not merged if you approved it?

**Pull request merged**